test/parallel/test-stream-pipe-objectmode-to-non-objectmode.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('node:assert');
+const { Readable, Transform } = require('node:stream');
+
+{
+  const objectReadable = Readable.from([
+    { hello: 'hello' },
+    { world: 'world' },
+  ]);
+
+  const passThrough = new Transform({
+    transform(chunk, _encoding, cb) {
+      this.push(chunk);
+      cb(null);
+    },
+  });
+
+  passThrough.on('error', common.mustCall());
+
+  objectReadable.pipe(passThrough);
+
+  assert.rejects(async () => {
+    // eslint-disable-next-line no-unused-vars
+    for await (const _ of passThrough);
+  }, /ERR_STREAM_INCOMPATIBLE_OBJECT_MODE/).then(common.mustCall());
+}
```

Comment:
Please add a test that the error is correctly forwarded back if:

1. source is an object mode stream that only flows strings/buffer and the error is unrelated
2. source is a binary stream and dest is a binary stream

Reply:
both test cases have been added 👍